pkg/kms/localkms/localkms.go

```diff
@@ -0,0 +1,195 @@
+	}
+
+	// register kmsClient in Tink
+	registry.RegisterKMSClient(kmsClient)
+
+	backendAEAD, err := kmsClient.GetAEAD(masterKeyURI)
+	if err != nil {
+		return nil, err
+	}
+
+	// create a KMSEnvelopeAEAD instance to wrap/unwrap keys managed by LocalKMS
+	masterKeyEnvAEAD := aead.NewKMSEnvelopeAEAD(*aead.AES256GCMKeyTemplate(), backendAEAD)
+
+	return &LocalKMS{store: s, secretLock: secretLock, masterKeyURI: masterKeyURI, masterKeyEnvAEAD: masterKeyEnvAEAD}, nil
+}
+
+// Create a new key/keyset for key type kt, store it and return its stored ID and key handle
+func (l *LocalKMS) Create(kt string) (string, interface{}, error) {
+	if kt == "" {
+		return "", nil, fmt.Errorf("failed to create new key, missing key type")
+	}
+
+	keyTemplate, err := getKeyTemplate(kt)
+	if err != nil {
+		return "", nil, err
+	}
+
+	kh, err := keyset.NewHandle(keyTemplate)
+	if err != nil {
+		return "", nil, err
+	}
+
+	kID, err := l.storeKeySet(kh)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return kID, kh, nil
+}
+
+// Get key handle for the given keyID
+func (l *LocalKMS) Get(keyID string) (interface{}, error) {
+	return l.getKeySet(keyID)
+}
+
+// Rotate a key referenced by keyID and return its updated handle
+func (l *LocalKMS) Rotate(kt, keyID string) (string, interface{}, error) {
+	kh, err := l.getKeySet(keyID)
+	if err != nil {
+		return "", nil, err
+	}
+
+	keyTemplate, err := getKeyTemplate(kt)
+	if err != nil {
+		return "", nil, err
+	}
+
+	km := keyset.NewManagerFromHandle(kh)
+
+	err = km.Rotate(keyTemplate)
+	if err != nil {
+		return "", nil, err
+	}
+
+	updatedKH, err := km.Handle()
+
+	if err != nil {
+		return "", nil, err
+	}
+
+	err = l.store.Delete(keyID)
+	if err != nil {
+		return "", nil, err
+	}
+
+	newID, err := l.storeKeySet(updatedKH)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return newID, updatedKH, nil
+}
+
+// nolint:gocyclo
+func getKeyTemplate(keyType string) (*tinkpb.KeyTemplate, error) {
+	switch keyType {
+	case "AES128GCM":
+		return aead.AES128GCMKeyTemplate(), nil
+	case "AES256GCMNoPrefix":
+		// RAW (to support keys not generated by Tink)
+		return aead.AES256GCMNoPrefixKeyTemplate(), nil
+	case "AES256GCM":
+		return aead.AES256GCMKeyTemplate(), nil
+	case "ChaCha20Poly1305":
+		return aead.ChaCha20Poly1305KeyTemplate(), nil
+	case "XChaCha20Poly1305":
+		return aead.XChaCha20Poly1305KeyTemplate(), nil
+	case "ECDSAP256":
+		return signature.ECDSAP256KeyTemplate(), nil
+	case "ECDSAP384":
+		return signature.ECDSAP384KeyTemplate(), nil
+	case "ECDSAP521":
+		return signature.ECDSAP521KeyTemplate(), nil
+	case "ED25519":
+		return signature.ED25519KeyTemplate(), nil
+	default:
+		return nil, fmt.Errorf("key type unrecognized")
+	}
+}
+
+func (l *LocalKMS) storeKeySet(kh *keyset.Handle) (string, error) {
+	w := newWriter(l.store, l.masterKeyURI)
+
+	buf := new(bytes.Buffer)
+	jsonKeysetWriter := keyset.NewJSONWriter(buf)
+
+	if e := kh.Write(jsonKeysetWriter, l.masterKeyEnvAEAD); e != nil {
```

Comment:
Can we be consistent and use err

And 

if err != nil

Reply:
done